027-network-policy-generation-feature-gate.md

```diff
@@ -0,0 +1,63 @@
+# Network Policy Generation Feature Gate
+
+The feature would consist of a new Feature Gate which can be used to disable the generation of Network Policies by Strimzi, where the user wishes to set up a more restrictive Network Policy set.
+
+## Current situation
+
+Currently, Strimzi generates Network Policies which allow all pods in the deployment to connect to Kafka, Zookeeper and any other service that it has generated.
+
+There is no way to stop this generation.
+
+It is possible, in the select case of listeners, to further restrict the Ingress using NetworkPolicyPeers, however this does not extend to all Network Policy generation, for example Metrics.
+It is also not possible to set up Egress Network Policies. 
+
+## Motivation
+
+The current Network Policies that are generated are great for getting Strimzi up and running in a deployment where a deny-all network policy is in place.
+
+However, this does not allow users of Strimzi to reach a fully secured Network Policy set, due to Strimzi not providing the full feature set of Kubernetes Network Policies.
+
+By adding a feature gate, it allows users to disable this generation, and instead write fully secure Network Policies.
+
+This setup is valid in 2 situations:
+
+* A deployment with no deny-all Network Policy, where all pods can already communicate with each other.
+* A deployment where the consumer wishes to write their own secure Network Policies.
+
+## Proposal
+
+The proposal would be to introduce a Feature Gate which would provide the ability to turn of the generation of Network Policies by Strimzi.
+
+As the functionality already exists to generate Network Policies, a sensible naming of this gate would be `NetworkPolicyGeneration` which by default is set to `true`.
```

Comment:
I wonder if just `NetworkPolicies` would be better name for the feature gate. Or maybe we can turn it around and call it `DisableNetworkPolicies` and have it disabled by default? Let's see what others think.

Reply:
Happy to switch to `NetworkPolicies`, the "Generation" part is superfluous here as everything is generation in this context.
I am less of a fan of `DisableNetworkPolicies` as Network Policies are a feature, Disable Network Policies is a setting, but not overly fussed as long as the ability to turn these off is available.
I will switch to `NetworkPolicies` for now, and it can change again based on further feedback.

Reply:
Tbh I am not a fan of "negative" logic so I am more for `NetworkPolicies` defaulted to `true` than a `DisableNetworkPolicies` defaulted to false.

Reply:
I switched to `NetworkPolicies`